### DISCUSSION TITLE
Accessing tiff's bits per sample

### DISCUSSION THREAD

**Discussion by MathemanFlo** [libvips/libvips#3321] [Ideas]
libvips already allows accessing important image properties such as width, height, resolution, number of channels, as well as some metadata. 
A typical requirement in my pipelines is to load (tif) images with variying bits per sample values (1,2,4,8), process them and save them later on with the same bits per sample (using bitdepth while saving). 
As far as I understand this is currently not possible as the bits per sample tag is read while parsing the tiff https://github.com/libvips/libvips/blob/d121ac229f410a0ae6bab9e8a46e0dfe8064db85/libvips/foreign/tiff2vips.c#L2780

but not stored such that it can be accessed by api in

https://github.com/libvips/libvips/blob/d121ac229f410a0ae6bab9e8a46e0dfe8064db85/libvips/foreign/tiff2vips.c#L1663

I would be very interested in a feature to access this property and willing to implement it as well. How should I do it?
Should I introduce a `VIPS_META_BITS_PER_SAMPLE`? and set it? I believe this property to be interesting for other image format, too...
Thank you very much for your help in advance.

**Comment by jcupitt** [comment]
Hi @MathemanFlo,

Sure, this sounds like a useful feature, a PR would be welcome. At least PNG, GIF and WEBP could also potentially support this, I think.

I've made an enhancement issue to track this.

**Reply by jcupitt** [reply]
https://github.com/libvips/libvips/issues/3322
